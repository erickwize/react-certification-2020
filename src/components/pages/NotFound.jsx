import React from "react";
import styled from "styled-components";
import { StyledTitle } from "../VideoList/VideoList";
const StyledContainer = styled.div`
  background-color: ${(props) => props.theme.background};
  color: ${(props) => props.theme.primaryText};
  display: flex;
  justify-content: center;
  position: fixed;
  width: 100vw;
  width: 100vh;
  opacity: 0.6;
`;
const StyledParagraph = styled.h1`
  font-family: "Roboto", sans-serif;
  font-size: 16px;
  color: ${(props) => props.theme.primaryText};
  margin-bottom: 24px;
`;
const NotFound = () => {
  return (
    <StyledContainer>
      <StyledTitle>Not Found</StyledTitle>
      <StyledParagraph>
        The page you are looking for does not exist...
      </StyledParagraph>
    </StyledContainer>
  );
};

export default NotFound;
