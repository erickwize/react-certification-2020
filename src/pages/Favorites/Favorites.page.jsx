import React from 'react';

import { HomeSection, Title, VideoContainer } from '../Home/Home.styled';
import { VideoCard, Tooltip } from '../../components';
import { useGlobalProvider } from '../../store/global/global.provider';
import { removeVideo } from '../../store/global/GlobalAction';

const Favorites = () => {
  const {
    state: { favoriteVideos, user },
    dispatch,
  } = useGlobalProvider();

  const removeFavorite = (videoId) => {
    const newFavorites = favoriteVideos.filter((video) => video.videoId !== videoId);
    removeVideo(dispatch, newFavorites);
  };

  if (favoriteVideos.length === 0)
    return (
      <HomeSection>
        <Title>
          <h1>Add videos to your favorite list</h1>
        </Title>
      </HomeSection>
    );

  return (
    <>
      <HomeSection>
        <Title>
          <h1>Welcome to your favorite list!</h1>
        </Title>
        <VideoContainer>
          {favoriteVideos.map((video) => {
            return (
              <Tooltip key={video.videoId}>
                {(props) => (
                  <VideoCard
                    {...props}
                    data={{ ...video, favoriteVideos, user }}
                    handlers={{ removeFavorite }}
                  />
                )}
              </Tooltip>
            );
          })}
        </VideoContainer>
      </HomeSection>
    </>
  );
};

export default Favorites;
