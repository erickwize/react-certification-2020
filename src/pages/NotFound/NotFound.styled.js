import styled from 'styled-components';

const Section = styled.section`
  display: flex;
  align-items: center;
  flex-direction: column;
  min-height: 100vh;
  padding: 20px 40px;

  a {
    font-size: 1rem;
    text-decoration: none;
    display: block;
    text-align: right;
    padding: 0.2rem 0.3rem;

    ::before {
      content: '‚Üê';
      padding-right: 0.2rem;
      display: inline-block;
    }
  }
`;

const Title = styled.h1`
  flex-basis: 100%;
  text-align: center;
  font-size: 3rem;
  letter-spacing: -2px;
  margin: 10px 20px;
`;

const Imagen = styled.img`
  width: 360px;
  height: 400px;
  margin: 10px 20px;
`;

export { Section, Title, Imagen };
