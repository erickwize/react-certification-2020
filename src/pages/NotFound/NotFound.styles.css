.not-found h4{
  color: red;
  font-size: 100px;
  display: block;
  margin: auto;
  text-align: center;
}
.not-found h5{
  color: black;
  display: block;
  margin: auto;
  font-size: 25px;
  text-align: center;
}

.not-found .home-link {
  color: black;
  display: block;
  margin: auto;
  font-size: 25px;
  text-align: center;
}

.not-found .home-link::before {
  content: '‚Üê';
  padding-right: 0.2rem;
  display: inline-block;
}

.not-found .image404 img{
  display: block;
  margin: auto;
  width: 200px;
  margin-top: 40px;
}

.dark404 h5{
  color: rgb(220, 220, 220);
}

.dark404 .home-link {
  color: rgb(220, 220, 220);
}

.dark404 .home-link::before {
  color: rgb(220, 220, 220);
}
