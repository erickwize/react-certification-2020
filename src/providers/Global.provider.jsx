import React, { useContext, useReducer } from 'react';
import { reducer } from '../reducer';
import { storage } from '../utils/storage';
import { dark } from '../Themes.styles';

const theme = JSON.parse(storage.get('theme')) || dark;
const user = JSON.parse(storage.get('user')) || {authenticated:false};
const favorites = storage.get('favorites') || [];

const GlobalContext = React.createContext(null);

function useGlobal() {
  const context = useContext(GlobalContext);
  if (!context) {
    throw new Error(`Can't use "useGlobal" without an GlobalProvider!`);
  }
  return context;
}

function GlobalProvider({children}) {

  const [state, dispatch] = useReducer(reducer, {theme, search:'doggos', user, favorites, isMenuOpen:false});

  return (
    <GlobalContext.Provider value={{state, dispatch}}>
      {children}
    </GlobalContext.Provider>
  );
}

export { useGlobal };
export default GlobalProvider;