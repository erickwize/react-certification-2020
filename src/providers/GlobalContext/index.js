export {
  initalState,
  GlobalContext,
  useVideList,
  useVideoInfo,
  reducer,
} from './GlobalContext.provider';
