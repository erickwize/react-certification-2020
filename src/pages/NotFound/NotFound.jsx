import React from 'react';
import { Link } from 'react-router-dom';

import './NotFound.css';

function NotFound() {
  return (
    <section className="not-found">
      <Link to="/" className="home-link">
        home
      </Link>
      <img src="404.gif" alt="page not found" />
    </section>
  );
}

export default NotFound;
