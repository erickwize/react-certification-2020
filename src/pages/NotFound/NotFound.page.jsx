import React from 'react';
import { Link } from 'react-router-dom';

import { Section, Title, Imagen } from './NotFound.styled';

function NotFoundPage() {
  return (
    <Section className="not-found">
      <Title>Page not found</Title>
      <Imagen src="404.gif" alt="page not found" />
      <Link to="/" className="home-link">
        Return Home
      </Link>
    </Section>
  );
}

export default NotFoundPage;
