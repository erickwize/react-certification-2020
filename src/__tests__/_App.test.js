import React from 'react';
import { screen, cleanup, render, fireEvent } from '@testing-library/react';
import { renderHook, act } from '@testing-library/react-hooks';
import AppLayout from '../components/App';
import Favorites from '../pages/Favs';

import NotFound from '../pages/NotFound';
import { useVideoInfo, useVideList, reducer, initalState } from '../providers/GlobalContext';
import VideoMock from '../utils/mock/youtube-videos-mock.json';

afterEach(() => {
  cleanup();
});

const MockVideoInfo = VideoMock.items[1].snippet;
const videoDataMock = {
  channelId: 'UCPGzT4wecuWM0BH9mPiulXg',
  channelTitle: 'Wizeline',
  description:
    'Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...',
  liveBroadcastContent: 'none',
  publishTime: '2019-09-30T23:54:32Z',
  publishedAt: '2019-09-30T23:54:32Z',
  thumbnails: {
    default: {
      height: 90,
      url: 'https://i.ytimg.com/vi/nmXMgqjQzls/default.jpg',
      width: 120,
    },
    high: {
      height: 360,
      url: 'https://i.ytimg.com/vi/nmXMgqjQzls/hqdefault.jpg',
      width: 480,
    },
    medium: {
      height: 180,
      url: 'https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg',
      width: 320,
    },
  },
  title: 'Video Tour | Welcome to Wizeline Guadalajara',
};

it('Should render NotFound', () => {
  const notFoundPage = render(<NotFound />);
  expect(notFoundPage).toMatchSnapshot();
  const notfound = screen.getByTestId('not-found');
  expect(notfound).toBeInTheDocument();
});

it('Should render Favorites', () => {
  const favorites = render(<Favorites />);
  expect(favorites).toMatchSnapshot();
});

it('Should render App', () => {
  act(() => {
    const app = render(<AppLayout />);
    expect(app).toMatchSnapshot();
  });
  const headerMenu = screen.getByTestId('yt-header');
  expect(headerMenu).toBeInTheDocument();
});

it('Should render App Header', () => {
  act(() => {
    render(<AppLayout />);
  });
  const headerMenu = screen.getByTestId('yt-header');
  expect(headerMenu).toBeInTheDocument();
  const SearchInput = screen.getByPlaceholderText('Search...');
  expect(SearchInput).toBeInTheDocument();
  SearchInput.focus();
  fireEvent.change(SearchInput, { target: { value: 'rise against' } });
  expect(SearchInput.value).toBe('rise against');
  fireEvent.keyDown(SearchInput, { key: 'Enter', code: 'Enter' });
});

it('Test useVideoInfo Hook', () => {
  const { result } = renderHook(() => useVideoInfo());
  expect(result.current.video).toEqual({});
  expect(typeof result.current.updateVideoInfo).toBe('function');

  act(() => {
    result.current.updateVideoInfo(MockVideoInfo);
  });
  expect(result.current.video).toEqual(videoDataMock);
});

it('Test useVideList Hook', () => {
  const { result } = renderHook(() => useVideList());
  expect(result.current.videoList).toEqual([]);
  expect(typeof result.current.updateVideoList).toBe('function');
  act(() => {
    result.current.updateVideoList([
      VideoMock.items[0],
      VideoMock.items[1],
      VideoMock.items[3],
    ]);
  });
  const videoList = [VideoMock.items[0], VideoMock.items[1], VideoMock.items[3]];
  expect(result.current.videoList).toEqual(videoList);
});

it('Test Reducer function', () => {
  expect(reducer(initalState, {type: 'SET_HISTORY', payload: 'Shinee' })).toEqual({
    ...initalState,
    history: 'Shinee',
  });
  expect(reducer(initalState, {type: 'SET_LIGHT_MODE' })).toEqual(initalState);
  expect(reducer(initalState, {type: 'SET_DARK_MODE' })).toEqual({
    ...initalState,
    header: {
      background: '#fea79c',
      input: '#0e4b79'
    },
    card: {
      background: '#f9c7ae',
      textColor: '#0e4b79',
    },
    background: '#0e4b79'
  });
  expect(reducer(initalState, {type: 'SET_SEARCH_KEYWORD', payload: 'Glass Animals' }));
  expect(reducer(initalState, {type: 'SET_FAVS_LIST', payload: { 'id': videoDataMock }}));
  expect(reducer(initalState, {type: 'REMOVE_USER_INFO', payload: 'Glass Animals' })).toEqual(initalState);
  expect(reducer(initalState, {type: 'SET_USER_INFO', payload: {
    id: '123',
    name: 'Wizeline',
    avatarUrl:
      'https://media.glassdoor.com/sqll/868055/wizeline-squarelogo-1473976610815.png',
  }})).toEqual({
    ...initalState,
    user: {
      id: '123',
      name: 'Wizeline',
      avatarUrl:
        'https://media.glassdoor.com/sqll/868055/wizeline-squarelogo-1473976610815.png',
    }
  });
})

it('Test Reducer: unkown action', () => {
  try {
    expect(reducer(initalState, { type: 'UNKOWN_ACTION' }));
  } catch (e) {
    expect(e.message).toBe('ACTION NOT RECOGNIZED');
  }
})
