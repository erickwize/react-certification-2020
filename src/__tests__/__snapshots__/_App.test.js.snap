// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bdfBwQ hIHNKi"
        data-testid="yt-header"
      >
        <div
          class="sc-gsTCUz ywWHk"
        >
          <button
            class="sc-dlfnbm cOuLkH"
          />
          <div
            class="sc-hKgILt ljSDGR"
          >
            <input
              class="sc-eCssSg iBGkVj"
              id="name"
              placeholder="Search..."
              type="text"
              value="wizeline"
            />
            <span
              class="sc-jSgupP hcyBed"
            />
          </div>
        </div>
        <div
          class="sc-gsTCUz ywWHk"
        >
          <div
            class="sc-gKsewC kjfNIn"
          >
            <input
              class="sc-pFZIQ gGzfcK"
              id="darkMode"
              name="darkMode"
              type="checkbox"
            />
            <label
              class="sc-iBPRYJ cpTqSp"
              for="darkMode"
            >
              Dark mode
            </label>
          </div>
          <button
            class="sc-fubCfw dDcTFR"
          />
        </div>
      </header>
      <main
        class="container"
        style="background: rgb(255, 255, 255);"
      >
        <h3
          data-testid="loading"
        >
          <br />
          Loading...
        </h3>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bdfBwQ hIHNKi"
      data-testid="yt-header"
    >
      <div
        class="sc-gsTCUz ywWHk"
      >
        <button
          class="sc-dlfnbm cOuLkH"
        />
        <div
          class="sc-hKgILt ljSDGR"
        >
          <input
            class="sc-eCssSg iBGkVj"
            id="name"
            placeholder="Search..."
            type="text"
            value="wizeline"
          />
          <span
            class="sc-jSgupP hcyBed"
          />
        </div>
      </div>
      <div
        class="sc-gsTCUz ywWHk"
      >
        <div
          class="sc-gKsewC kjfNIn"
        >
          <input
            class="sc-pFZIQ gGzfcK"
            id="darkMode"
            name="darkMode"
            type="checkbox"
          />
          <label
            class="sc-iBPRYJ cpTqSp"
            for="darkMode"
          >
            Dark mode
          </label>
        </div>
        <button
          class="sc-fubCfw dDcTFR"
        />
      </div>
    </header>
    <main
      class="container"
      style="background: rgb(255, 255, 255);"
    >
      <h3
        data-testid="loading"
      >
        <br />
        Loading...
      </h3>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Should render Favorites 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        data-testid="favorites-view"
      >
        <h1
          class="sc-kLgntA gYAoXa"
        >
          Your favorites
        </h1>
        <p>
          0
           favorite videos
        </p>
        <section
          class="sc-bYEvPH dCExnP"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      data-testid="favorites-view"
    >
      <h1
        class="sc-kLgntA gYAoXa"
      >
        Your favorites
      </h1>
      <p>
        0
         favorite videos
      </p>
      <section
        class="sc-bYEvPH dCExnP"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Should render NotFound 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="not-found"
        data-testid="not-found"
      >
        <a
          class="home-link"
          to="/"
        >
          home
        </a>
        <img
          alt="page not found"
          src="404.gif"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="not-found"
      data-testid="not-found"
    >
      <a
        class="home-link"
        to="/"
      >
        home
      </a>
      <img
        alt="page not found"
        src="404.gif"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
